### ISSUE TITLE
[BUG] Screenshot is created without scrollbars in scrollbar exists screen

### ISSUE THREAD

**Issue by calc29** [microsoft/playwright#29140]
### System info
- Playwright Version: v1.41.1
- Operating System: Windows 10
- Browser: Chromium
- Other info: nodejs

### Source code
html: [[code link](https://github.com/calc29/playwright-test/blob/main/scroll-test.html)]
js: [[code link](https://github.com/calc29/playwright-test/blob/main/scroll-test.js)]

**Steps**
- Run js code
- Goto html link url on Playwright browser
- Click screenshot button
    case1. scrollbar not exists screen
       ![image](https://github.com/microsoft/playwright/assets/142505476/b90b0e8d-e7df-4581-983a-ba7026cbdd22)
    case2. scrollbar exists screen
       ![image](https://github.com/microsoft/playwright/assets/142505476/b4865b02-bfaa-4161-83b5-0ee88e440f14)

### Describe the bug
If a screenshot is taken on a screen where scrollbars are present, the scrollbars in the textarea will disappear.
This is not a problem on screens where scrollbars do not exist.

**Expected**
Scrollbars in the textarea appear in both cases 1 and 2.

**Actual**
case1. screenshot result: scrollbar not exists screen
 no problem
![case1 scroll-bug-screenshot](https://github.com/microsoft/playwright/assets/142505476/264a2cff-441e-4681-a68e-3d36355cf748)

case2. screenshot result: scrollbar exists screen (bug)
 scrollbar is not rendered.
![case2 scroll-bug-screenshot](https://github.com/microsoft/playwright/assets/142505476/e31943f6-b2a9-4ca1-9403-11525494325c)

**Comment by yury-s** [comment]
I cannot reproduce the difference in the text area scroll bar behavior. As for the page scroll bar, it is expected when you take screenshot with `fullPage: true` rather than just the page viewport. If you remove fullPage parameter, the screenshot will contain what you see on the screen. Would that work for you?

**Comment by calc29** [comment]
Thank you for your response.
Try case2 after adjusting the screen height to allow for scroll bar. It should be reproducible.
If I remove `fullPage: true` because I want to take a screenshot of the entire page without setting viewport, I do not get the expected result. 
If I remove `fullPage: true` and setting the viewport, is there any way to change it back to `viewport: null` because it will no longer auto-adjust to fit the window when the browser is resized?

**Comment by yury-s** [comment]
I'm not sure I'm following what is not working. As I understand your description:

You create a context with vieport:null, then resize the browser window.
* If you take screenshot without extra options you get the scrollbar on the page as expected but you don't see full content as it doesn't fit into the viewport.
* If you take screenshot with `fullPage: true` then the scrollbar disappears as you capture entire content.
Both scenarios are working as intended.

> If I remove fullPage: true and setting the viewport, is there any way to change it back to viewport: null because it will no longer auto-adjust to fit the window when the browser is resized?

No, this is not supported. You can configure viewport only once when creating the context. However, you can call [page.setViewportSize](https://playwright.dev/docs/api/class-page#page-set-viewport-size) to resize the viewport.

**Comment by calc29** [comment]
I have no problem with the scrollbars disappearing on the screen itself, but isn't it a bug that the scrollbars on the <textarea> form disappear with it?
I would like the contents of the screen to match exactly the contents of the screenshot.

**Comment by alescinskis** [comment]
IIRC scrollbars are hidden by default, you can override this by adding following in playwright config file:
```
use: {
    launchOptions: {
      ignoreDefaultArgs: ['--hide-scrollbars'],
    },
```
keep in mind this will work only for chrome/chromium

**Comment by calc29** [comment]
> IIRC scrollbars are hidden by default, you can override this by adding following in playwright config file:
> 
> ```
> use: {
>     launchOptions: {
>       ignoreDefaultArgs: ['--hide-scrollbars'],
>     },
> ```
> 
> keep in mind this will work only for chrome/chromium

Thanks for your comment.
I have already applied that option.
But as I reported, the scrollbars in the text area are disappearing.
It is the same in chrome.

This is my code:
```
const browser = await playwright.chromium.launch({
        channel: 'msedge',
        headless: false,
        ignoreDefaultArgs: ['--hide-scrollbars']
    });
```

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
